index.js

```diff
@@ -250,28 +220,69 @@ function noteRecorder(note, duration, timeIn) {
     }
 }
 
-//SAVING FUNCTIONALITY
-const saveBtn = document.getElementById('save_song')
+//COPYING FUNCTIONALITY
+const songInputName = document.getElementById('song_name')
+const copyBtn = document.getElementById('copy_url')
+
+copyBtn.addEventListener('click', e => {
+    copyBtn.style="background:red;color:#fff;";
+    copyBtn.innerHTML = "Copying";
+    setTimeout(function() { copyBtn.style= ""; copyBtn.innerHTML = "Copy!";}, 1000);
+    currentSong = {name: songInputName.value, notes: newRecording}
+    
+    songToURL()
+    copyToClipboard(songLocation)
+    
+    songInputName.value = "Song Copied. Share With Your Friends!"
+    setTimeout(()=> songInputName.value = "Record Another Song!", 5000)
+})
 
-saveBtn.addEventListener('click',
-    function () {
-        let songName = document.getElementById('song_name')
+let songLocation
 
-        saveBtn.style="background:red;color:#fff;";
-        saveBtn.innerHTML = "Saving";
-        setTimeout(function() { saveBtn.style= ""; saveBtn.innerHTML = "Save"; songName.value = "Your song was saved! Check the list." }, 1000);
+function songToURL() {
+    songLocation = `${window.location.href.split("#")[0]}/#${encodeURIComponent(JSON.stringify(currentSong))}`
+    console.log(songLocation)
+}
 
-        currentSong = {name: songName.value, notes: newRecording}
-        postSong(currentSong)
-        
-        setTimeout(()=> songName.value = "", 5000)
-    }
-)
+function urlToSong() {
+    let uri = decodeURIComponent(window.location.href.split("#")[1])
```

Comment:
This can be a const

Reply:
Is there a case where this only makes a list of one element then this get index out of bound?